src/transport/SecurePairingSession.cpp

```diff
@@ -361,8 +358,8 @@ CHIP_ERROR SecurePairingSession::HandlePBKDFParamRequest(const PacketHeader & he
     ChipLogDetail(Ble, "Received PBKDF param request");
 
     // Update commissioning hash with the received pbkdf2 param request
-    err = mCommissioningHash.AddData(req, reqlen);
```

Comment:
We should keep this behaviour (hashing in the PBKDFParamRequest).

Reply:
Instead of passing the context in chunks, it would be better to pass complete context to crypto layer. Anyway we need a place holder to keep track of the length of the context ("CHIP PAKE V1 Commissioning" || PBKDFParamRequest || PBKDFParamResponse).

Reply:
If the crypto operations are offloaded to HSM, it would be convenient if the context is passed as a whole.

Reply:
I'm not suggesting we pass in the context in chunks. The context **is** the **hash** of the constant string "CHIP PAKE V1 Commissioning", the PBKDFParamRequest, and the PBKDFParamResponse. We implicitly know the length because we know the hash function that is used. 

**Previously Spake2+ was calculating this:** 

```
TranscriptHash = SHA256("Const String" || PBKDFParamRequest || PBKDFParamResponse || 
                        0x0000000000000000 || 0x0000000000000000 || 
                        len(M) || M || len(N) || N || 
                        len(X) || X || len(Y) || Y || 
                        len(Z) || Z || len(V) || V || len(w0) w0)
```

**What we would be calculating with this path:**

```
TranscriptHash = SHA256(len("Const String" || PBKDFParamRequest || PBKDFParamResponse) || 
                        "Const String" || PBKDFParamRequest || PBKDFParamResponse ||
                        0x0000000000000000 || 0x0000000000000000 || 
                        len(M) || M || len(N) || N || 
                        len(X) || X || len(Y) || Y || 
                        len(Z) || Z || len(V) || V || len(w0) w0)
```

**What we should be calculating:**

```
TranscriptHash = SHA256(SHA256_Output_Length || 
                        SHA256("Const String" || PBKDFParamRequest || PBKDFParamResponse) ||
                        0x0000000000000000 || 0x0000000000000000 || 
                        len(M) || M || len(N) || N || 
                        len(X) || X || len(Y) || Y || 
                        len(Z) || Z || len(V) || V || len(w0) w0)
```


This method of hashing these values is convenient as devices don't need to store the full-length message in memory. This will be especially important if we increase the number of pre-session messages or increase their size.